internal/oidc/session_test.go

```diff
@@ -133,7 +133,7 @@ func TestSessionGenerator(t *testing.T) {
 		require.NotEqual(t, sg.GenerateNonce(), sg.GenerateNonce())
 	})
 	t.Run("static", func(t *testing.T) {
-		sg := NewStaticGenerator("sessionid", "nonce", "state")
+		sg := NewStaticGenerator("sessionid", "nonce", "state", "codeverifier")
 		require.Equal(t, sg.GenerateSessionID(), sg.GenerateSessionID())
 		require.Equal(t, sg.GenerateState(), sg.GenerateState())
 		require.Equal(t, sg.GenerateNonce(), sg.GenerateNonce())
```

Comment:
[nit] Update this test and the random one to also include a check for the code verifier.

Reply:
@nacx : Thanks for the review. Yes, there is a way to configure keycloak to enforce PKCE use by client. Infact, I also used keycloak for my local testing. In the UI, It's available under Client > Advanced > Proof Key for Code Exchange Code Challenge Method > Set it to S256. (I am not sure if the same setting is available through the Keycloak admin API)

Here is a screenshot from the UI:
![image](https://github.com/user-attachments/assets/0f658cf2-e028-485e-bae8-d5ee96f9daa7)

Reply:
The client is created and configured here: https://github.com/istio-ecosystem/authservice/blob/main/e2e/keycloak/setup-keycloak.sh#L50-L58
Let's see if PKCE can be configured in that script.

Actually, there is probably no need to create another suite. The `istio` suite also uses Keycloak, so we can just change the `keycloak` suite and we should have all cases covered.

Testing ti locally should be easy:

```
make docker   # the e2e tests need the update Docker image
make e2e/keycloak

# Or, if you want to see all requests/responses for the e2e tests logged:
E2E_TEST_OPTS="-v -count=1" make e2e/keycloak
```

Reply:
Adding this to the keycloak setup script sould do the trick, to enforce PKCE server-side:
```diff
diff --git a/e2e/keycloak/setup-keycloak.sh b/e2e/keycloak/setup-keycloak.sh
index 7fc46af..ee3d340 100755
--- a/e2e/keycloak/setup-keycloak.sh
+++ b/e2e/keycloak/setup-keycloak.sh
@@ -51,6 +51,7 @@ set -ex
     -s clientId="${CLIENT_ID}" \
     -s secret="${CLIENT_SECRET}" \
     -s "redirectUris=[\"${REDIRECT_URL}\"]" \
+    -s "attributes={\"pkce.code.challenge.method\":\"S256\"}" \
     -s consentRequired=false \
     --server "${KEYCLOAK_SERVER}" \
     --realm "${REALM}" \
```

Reply:
Added and ran e2e tests, they run fine.

Reply:
Nice! I run the e2e with this setting but without the changes in the PR and they were failing, complaining about the missing query params, so all good!